### ISSUE TITLE
“TGA Type 2 (uncompressed RGB) top-bottom Test Node” devtest node texture is flipped upside down

### ISSUE THREAD

**Issue by erlehmann** [minetest/minetest#12570]
##### Minetest version
Commits 7c227d2a004068267dbe713259d57374a70c29e4 up to 2cf52642fa89a9497b0200f5a4637a25a82f9c3d

##### OS / Hardware
It does not matter, as this is not a rendering error.

##### Summary
The file `games/devtest/mods/testnodes/textures/testnodes_tga_type2_16bpp_tb.tga` is flipped upside down.

Possible reason: Converting from bottom-top to top-bottom pixel order using imagemagick can erroneously flip images.

##### Steps to reproduce
1. Place the TGA test nodes in devtest.
2. Notice that one of the nodes has a ttexture that is flipped upside down.

##### Additional Information
If you fix this by flipping the texture, make sure that the resulting image is actually a type 2 TGA with 16bpp (A1R5G5B5) for which imagemagick's identify(1) reports an orientation of `TopLeft`.

**Comment by Wuzzy2** [comment]
To be clear: So do you believe the error lies in the image itself?

**Comment by erlehmann** [comment]
> To be clear: So do you believe the error lies in the image itself?

Yes. No rendering bug, the texture is simply upside down due to Imagemagick developers acting like scanline order is display orientation information.

So most likely, when I inverted the scanline order, the image got flipped upside down.

**Comment by sfan5** [comment]
Since you contributed the TGA test files maybe you could open up a PR that fixes them?

**Comment by erlehmann** [comment]
As soon as I figure out how to stop Imagemagick from flipping stuff erroneously, yes.

Or maybe I should just redraw it.

**Comment by erlehmann** [comment]
For future reference:

* Sometimes Imagemagick erroneously flips images upside down.  This is a bug.
* Imagemagick authors claim that users forgot the `-auto-orient` CLI option.
* I think `-auto-orient`  actually triggers the Imagemagick flip bug *twice*.
* So the Imagemagick authors' advice is right for the wrong reasons.

**Comment by erlehmann** [comment]
I intend to fix this bug as a side effect of regenerating the test textures. See https://github.com/minetest/minetest/issues/12840#issuecomment-1763418853 for details.
